openedx/features/job_board/tests/views/test_job_board.py

```diff
@@ -53,19 +72,154 @@ def test_create_view_post_unsuccessful(self):
+
+        response = self.client.get(reverse('job_list') + '?' + JOB_COMP_SALARIED_KEY + '=' + JOB_PARAM_TRUE_VALUE)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(response.context_data['search_fields'][JOB_COMP_SALARIED_KEY], JOB_PARAM_TRUE_VALUE)
+        self.assertTrue(response.context_data['filtered'], True)
+        self.assertTrue(len(response.context_data['job_list']) == number_of_salaried_jobs)
+        for job in response.context_data['job_list']:
+            self.assertEqual(job.compensation, JOB_COMP_SALARIED_KEY)
+
+    def test_job_list_view_filters_job_hours(self):
+        number_of_fulltime_jobs = 0
+        number_of_parttime_jobs = 0
+        number_of_freelance_jobs = 0
+
+        for job in self.jobs:
+            if job.hours is JOB_HOURS_FULLTIME_KEY:
+                number_of_fulltime_jobs += 1
+            if job.hours is JOB_HOURS_PARTTIME_KEY:
+                number_of_parttime_jobs += 1
+            if job.hours is JOB_HOURS_FREELANCE_KEY:
+                number_of_freelance_jobs += 1
+
+        response = self.client.get(reverse('job_list') + '?' + JOB_HOURS_FULLTIME_KEY + '=' + JOB_PARAM_TRUE_VALUE)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(response.context_data['search_fields'][JOB_HOURS_FULLTIME_KEY], JOB_PARAM_TRUE_VALUE)
+        self.assertTrue(response.context_data['filtered'], True)
+        self.assertTrue(len(response.context_data['job_list']) == number_of_fulltime_jobs)
+        for job in response.context_data['job_list']:
+            self.assertEqual(job.hours, JOB_HOURS_FULLTIME_KEY)
+
+        response = self.client.get(reverse('job_list') + '?' + JOB_HOURS_PARTTIME_KEY + '=' + JOB_PARAM_TRUE_VALUE)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(response.context_data['search_fields'][JOB_HOURS_PARTTIME_KEY], JOB_PARAM_TRUE_VALUE)
+        self.assertTrue(response.context_data['filtered'], True)
+        self.assertTrue(len(response.context_data['job_list']) == number_of_parttime_jobs)
+        for job in response.context_data['job_list']:
+            self.assertEqual(job.hours, JOB_HOURS_PARTTIME_KEY)
+
+        response = self.client.get(reverse('job_list') + '?' + JOB_HOURS_FREELANCE_KEY + '=' + JOB_PARAM_TRUE_VALUE)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(response.context_data['search_fields'][JOB_HOURS_FREELANCE_KEY], JOB_PARAM_TRUE_VALUE)
+        self.assertTrue(response.context_data['filtered'], True)
+        self.assertTrue(len(response.context_data['job_list']) == number_of_freelance_jobs)
+        for job in response.context_data['job_list']:
+            self.assertEqual(job.hours, JOB_HOURS_FREELANCE_KEY)
+
+    def test_job_list_view_filters_job_location(self):
+        job = JobFactory(country='PK', city='Karachi')
+        response = self.client.get(reverse(
```

Comment:
Lets add comment that job search filter was applied on what type of data, meaning 11 jobs (10 created in setup)

Please do the same for `test_job_list_view_filters_job_query`